### DISCUSSION TITLE
Save the model with best results

### DISCUSSION THREAD

**Discussion by ryabs07** [lululxvi/deepxde#1383] [Q&A]
How do we save the neural network model that yeilded best results throughout all the iterations?

For instance, the following code saves the model of **final** iteration which might _not necessarily_ have been the best model:
```
model = dde.Model(data, net)
dde.optimizers.set_LBFGS_options(maxiter= iterations, gtol= 1e-6)
pde_resampler = dde.callbacks.PDEPointResampler(period=10)
model.compile("L-BFGS", loss_weights = [1, 10, 10000, 10000]) #
losshistory, train_state = model.train(iterations=iterations, callbacks=[pde_resampler])
dde.saveplot(losshistory, train_state, issave=True, isplot=True)
model.save("LBFGS_model.h5")
```

**Comment by lululxvi** [comment]
Use dde.callbacks.ModelCheckpoint https://deepxde.readthedocs.io/en/latest/modules/deepxde.html#deepxde.callbacks.ModelCheckpoint
